robot_smach_states/test/navigation/constraint_functions/test_waypoint_constraints.py

```diff
@@ -37,20 +44,26 @@ def test_no_look(self):
         x_coordinate = 2.1
         y_coordinate = 3.7
         z_coordinate = 0
+        radius = 0.3
         yaw = 1.57
         frame_id = "/map"
 
         pose = kdl.Frame(kdl.Rotation.RPY(0, 0, yaw), kdl.Vector(x_coordinate, y_coordinate, z_coordinate))
         e = Entity("dummy", "dummy_type", frame_id, pose, None, None, None, None)
         entity = Designator(e, name="entity designator")
 
-        pc, oc = waypoint_constraint(entity, 0.3, look=False)
+        pc, oc = waypoint_constraint(entity, radius, look=False)
 
         # verify positionconstraint
-        self.assertIn("x", pc.constraint)
-        self.assertIn("y", pc.constraint)
+        constraint_string = pc.constraint
+        constraint_string = constraint_string.replace("^", "**")
         self.assertEqual(pc.frame, frame_id)
 
+        verification_string = "(x-{})**2+(y-{})**2 < {}**2".format(x_coordinate, y_coordinate, radius)
```

Comment:
Nice! 

Maybe it's handy to put code to check the constraint in a separate function that is given the verification_string and optionally a specification of the range we want to check it over (those linspaces). 
I'd then pass that verification_string in the same format as the original, so do the `^` to `**` on both. 

Then here you can just call `self.assertTrue(constraint_outcomes_equal(constraint_string, verification_string))` and it's super clear what is going on.
In that `constraint_outcomes_equal` (or a better name) you'll haveto document all the reasons why this is the way it's done and not eg. the string comparison I proposed and the reasons why that wouldn't work. Because that question will pop up for other reviewers and later.

Reply:
Seperated the comparison function. However I am not sure that comparing a matrix of evenly spaced points is the best way to go about this. Because right now we test points that are 10cm apart, which is a lot for some position constraints. I think it might be better to test a handful of strategically chosen points rather than a coarse array of effectively random points

Reply:
We could consider https://stackoverflow.com/questions/37112738/sympy-comparing-expressions but that's overkill I guess. 
Going down to comparing at a higher resolution gives more details of course, but I think these unit tests should mostly tell if nothing broke really. If that case, the outlines of the functions will be wildly different I expect.